storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/connection/SnowflakeConnectionProvider.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.storage.connectors.jdbc.connection;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Properties;
+
+public class SnowflakeConnectionProvider implements JdbcConnectionProvider {
+  private final String database;
+  private final String schema;
+  private final String password;
+  private final String username;
+  private final String url;
+  private final String className;
+
+  public SnowflakeConnectionProvider(Map<String, String> config) {
+    if (!config.containsKey("database")
```

Comment:
this validation logic is incomplete and would be better delegated to the driver which has more config settings than this.

Reply:
So, are you saying we should just let the `DriverManager` handle the validation logic?

Reply:
not DriverManager but the snowflake driver implementation already does, yes.  You only need to catch the exceptions from the conn init, if there is something we can do about it.